### PR TITLE
Use proxy to bypass adventure.land CORS policy

### LINKED ISSUE #3: [Bug] Gear Planner - Import characters CORS issues in production build
**Describe the bug**
When you try to import one or more characters, no data is returned due to CORS issues. setupProxy.js only works during development

**Expected behavior**
Characters should be extracted out of adventure.land so you can select them in the UI.

### PR DESCRIPTION
**Pull request by kevinsandow**
Solves #3 

Both proxy functions are publicly available here:

https://dash.deno.com/playground/adventureland-character
https://dash.deno.com/playground/adventureland-player

### CODE REVIEW

**Comment by thmsndk** [comment]
I'll review the code later today when I have some more time, one thing though, I'd like for the proxy to be disabled during development and only run in production.

Should be able to configure that in the replace part of [setupProxy.js](https://github.com/adventureland-community/al-data-explorer/blob/main/src/setupProxy.js) so it does not hit the deno endpoint, but just calls adventureland directly.

**Comment by kevinsandow** [comment]
I don't know actually know why this should be a feature. In case the proxy no longer works?

**Comment by thmsndk** [comment]
> I don't know actually know why this should be a feature. In case the proxy no longer works?

Because there is no reason to send it to another proxy, when the local development proxy is enough :) 
It also removes a dependency for the people who are running it locally in development mode.

**Comment by kevinsandow** [comment]
I re-added the possibility to use adventure.land directly when running in local development mode.

**Review by thmsndk — APPROVED** [review]

File: src/GearPlanner/useIsLocalEnvironment.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useMemo } from "react";
+
+const localHostnames = [
+    'localhost',
+    '127.0.0.1',
+    '::1',
+]
+
+export default () => useMemo(() => {
+    return localHostnames.includes(location.hostname)
```
Comment (Telokis) [inline src/GearPlanner/useIsLocalEnvironment.ts]: Rely on `process.env.NODE_ENV !== "production"`, instead. No need to check the location at all.

**Review by Telokis — CHANGES_REQUESTED** [review]

File: src/GearPlanner/useIsLocalEnvironment.ts (hunk 1 above)
Comment (kevinsandow) [reply src/GearPlanner/useIsLocalEnvironment.ts]: I changed it accordingly, wasn't sure whether that would be available in a build. But I guess webpack makes it available.

**Pull request merged**